atlas-chart/src/main/scala/com/netflix/atlas/chart/util/PngImage.scala

```diff
@@ -87,12 +89,22 @@ object PngImage {
 
   def userError(imgText: String, width: Int, height: Int): PngImage = {
     val userErrorYellow = new Color(0xFF, 0xCF, 0x00)
-    error(imgText, width, height, "USER ERROR:", Color.BLACK, userErrorYellow)
+    val clampedWidth = math.min(width, GraphConstants.MaxWidth)
+    val clampedHeight = math.min(height, GraphConstants.MaxHeight)
+    
+    val dimensionWarning = if (width > GraphConstants.MaxWidth || height > GraphConstants.MaxHeight) {
+      s" Image dimensions clamped to ${clampedWidth}x${clampedHeight} (max: ${GraphConstants.MaxWidth}x${GraphConstants.MaxHeight})."
+    } else ""
+    
+    val fullMessage = imgText + dimensionWarning
+    error(fullMessage, clampedWidth, clampedHeight, "USER ERROR:", Color.BLACK, userErrorYellow)
   }
 
   def systemError(imgText: String, width: Int, height: Int): PngImage = {
     val systemErrorRed = new Color(0xF8, 0x20, 0x00)
-    error(imgText, width, height, "SYSTEM ERROR:", Color.WHITE, systemErrorRed)
+    val clampedWidth = math.min(width, GraphConstants.MaxWidth)
+    val clampedHeight = math.min(height, GraphConstants.MaxHeight)
```

Comment:
I would suggest moving this to the `error` method so it covers all the cases in a single place.

Reply:
Thanks for pointing this out! Will do

Reply:
This has ben moved to error()